app/vmalert/rule/alerting.go

```diff
@@ -47,13 +47,19 @@ type AlertingRule struct {
 }
 
 type alertingRuleMetrics struct {
-	errors        *utils.Gauge
+	errors        *utils.Counter
```

Comment:
I think it would be better to keep both types of metrics for a while for the sake of backwards compatibility.
Let's have both: counter and gauge. But gauge marked as deprecated in code comment.

The same goes for alerting rule and dashboard:
1. We need to add new alerting rule to use new metric instead of modifying the old one
2. We need to add new expression to dashboard panel instead of modifying existing one
This would make both alerts and dasbhoard compatible with different vmalert versions.

@Haleygo please validate my thinking here.

Reply:
@hagen1778 I think compatibility sounds nice, but it brings:
1. duplicated rules[**duplicated alerts!**]
2. duplicated panels
   _I assume we won't write rules or panels with expression using two metrics here._
3. double series data, for users who have a lot of rules, it's not good

>The same goes for alerting rule and dashboard:
We need to add new alerting rule to use new metric instead of modifying the old one
We need to add new expression to dashboard panel instead of modifying existing one
This would make both alerts and dasbhoard compatible with different vmalert versions.

I think it might bring more unhappy users. Since this version of rules and dashboards will only work 100% with a few versions of vmalert. So when we finally drop the deprecated metrics after few versions, upgrading vmalert without updating dashboards&rules accordingly will effect users who:
1. use rather old version and bump to the latest
2. use the version with both metrics 

Then what's the difference between dropping it now and keeping it longer.

Of course, if keeping the old one for a while is the rule of compatibility, I'm happy to learn)

Reply:
Hi @Haleygo , @hagen1778 ! I'm not sure about the alerting rules. I don't want to use both of them because it can annoy users. wdt?

Reply:
Agree